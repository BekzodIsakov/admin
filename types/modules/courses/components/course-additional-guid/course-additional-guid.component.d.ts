/**
 * <CourseAdditionalGuid />
 */
export const CourseAdditionalGuid: any;
//# sourceMappingURL=course-additional-guid.component.d.ts.map