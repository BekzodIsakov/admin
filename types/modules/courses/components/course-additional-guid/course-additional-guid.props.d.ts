export function useCourseAdditionalGuidProps(_: any): {
    filterOptions: {
        label: string;
        value: string;
    }[];
    additionalFiles: any;
};
//# sourceMappingURL=course-additional-guid.props.d.ts.map