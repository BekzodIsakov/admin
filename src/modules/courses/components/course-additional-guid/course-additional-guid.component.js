import React from 'react';
import { hoc } from '@utils/hoc';
import { useCourseAdditionalGuidProps } from './course-additional-guid.props';
import {
  CourseAdditionalGuidContainer,
  CourseAdditionalGuidTop,
  CourseAdditionalGuidTitle,
  CourseAdditionalGuidTopRight,
  CourseAdditionalGuidSubTitle,
  CourseAdditionalGuidBottom,
  // CourseAdditionalGuidCardImg,
  CourseAdditionalGuidCardWrapper
} from './course-additional-guid.style';
import { DropDownComponent } from '@components/drop-down/drop-down.component';
import { Card } from '../course-tasks/components/card/card.component';
import { SvgIcon } from '@components/atoms/index';
/**
 * <CourseAdditionalGuid />
 */
const CourseAdditionalGuid = hoc(
  useCourseAdditionalGuidProps,
  ({ filterOptions, additionalFiles }) => (
    <CourseAdditionalGuidContainer>
      {additionalFiles?.length > 0 && (
        <CourseAdditionalGuidTop>
          <CourseAdditionalGuidTitle>
            {additionalFiles.length} ta qo'shimcha qo'llanma va fayl mavjud
          </CourseAdditionalGuidTitle>

          <CourseAdditionalGuidTopRight>
            <CourseAdditionalGuidSubTitle>
              Saralash:
            </CourseAdditionalGuidSubTitle>
            <DropDownComponent
              defaultOption='type-file'
              options={filterOptions}
            />
          </CourseAdditionalGuidTopRight>
        </CourseAdditionalGuidTop>
      )}

      <CourseAdditionalGuidBottom>
        {additionalFiles?.length > 0 ? (
          additionalFiles.map((item, index) => (
            <CourseAdditionalGuidCardWrapper
              href={item?.file?.url}
              key={index}
              target='_blank'
            >
              <Card title={item.title} subtitle={item.file.size} large={true}>
                <SvgIcon iconName='download-icon' iconColor='#BA8D5B' />
              </Card>
            </CourseAdditionalGuidCardWrapper>
          ))
        ) : (
          <h4>Qoshimcha fayl mavjud emas</h4>
        )}
      </CourseAdditionalGuidBottom>
    </CourseAdditionalGuidContainer>
  )
);

export { CourseAdditionalGuid };
