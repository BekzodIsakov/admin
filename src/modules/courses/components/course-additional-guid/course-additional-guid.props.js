/**
 * <CourseAdditionalGuid /> props
 */

import { useSelector } from 'react-redux';

export const useCourseAdditionalGuidProps = _ => {
  const additionalFiles = useSelector(
    state => state.courses?.activeLesson?.resources
  );

  const filterOptions = [{ label: 'fayl turi', value: 'type-file' }];

  return {
    filterOptions,
    additionalFiles
  };
};
