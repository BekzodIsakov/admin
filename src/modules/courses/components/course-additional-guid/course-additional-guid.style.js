import styled from 'styled-components';

export const CourseAdditionalGuidContainer = styled.div``;

export const CourseAdditionalGuidTop = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin-bottom: 16px;
`;

export const CourseAdditionalGuidTitle = styled.h3`
  font-style: normal;
  font-weight: 700;
  font-size: 16px;
  line-height: 20px;
  letter-spacing: -0.02em;
  color: #000000;

  @media only screen and (max-width: 1024px) {
    max-width: 70%;
  }
`;

export const CourseAdditionalGuidTopRight = styled.div`
  display: flex;
  align-items: center;
`;

export const CourseAdditionalGuidSubTitle = styled.h4`
  font-style: normal;
  font-weight: 500;
  font-size: 14px;
  line-height: 18px;
  letter-spacing: -0.02em;
  color: rgba(0, 0, 0, 0.35);
  margin-right: 12px;

  @media only screen and (max-width: 1024px) {
    display: none;
  }
`;

export const CourseAdditionalGuidBottom = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-left: -10px;
  margin-right: -10px;
`;

export const CourseAdditionalGuidCardImg = styled.img`
  width: 44px;
  height: 44px;
  object-fit: cover;
`;

export const CourseAdditionalGuidCardWrapper = styled.a`
  padding: 0 10px;
  flex-basis: 50%;
  max-width: 50%;
  text-decoration: none;
  cursor: pointer;
  &:not(:last-child) {
    margin-bottom: 24px;
  }

  &:not(:nth-last-child(2)) {
    margin-bottom: 24px;
  }
`;
