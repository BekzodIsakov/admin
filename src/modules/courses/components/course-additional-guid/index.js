export * from './course-additional-guid.component';
