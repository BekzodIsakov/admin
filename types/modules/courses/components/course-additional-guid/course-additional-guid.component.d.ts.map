{"version":3,"file":"course-additional-guid.component.d.ts","sourceRoot":"","sources":["../../../../../src/modules/courses/components/course-additional-guid/course-additional-guid.component.js"],"names":[],"mappings":"AAgBA;;GAEG;AACH,uCA0CE"}