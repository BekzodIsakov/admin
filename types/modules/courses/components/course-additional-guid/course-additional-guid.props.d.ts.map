{"version":3,"file":"course-additional-guid.props.d.ts","sourceRoot":"","sources":["../../../../../src/modules/courses/components/course-additional-guid/course-additional-guid.props.js"],"names":[],"mappings":"AAMO;;;;;;EAWN"}