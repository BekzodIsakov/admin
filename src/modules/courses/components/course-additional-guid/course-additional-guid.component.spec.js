import { customRender } from '@test-utils/helpers';
// import { screen } from "@testing-library/react";

import { CourseAdditionalGuid } from './course-additional-guid.component';

const setup = () => {
  return customRender(<CourseAdditionalGuid />);
};

test('Course additional guide test', () => {
  const { asFragment } = setup({});
  expect(asFragment).toMatchSnapshot();
});
