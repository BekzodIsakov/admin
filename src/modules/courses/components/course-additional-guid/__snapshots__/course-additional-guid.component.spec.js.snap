// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Course additional guide test 1`] = `[Function]`;
